fga/src/client.rs

```diff
@@ -0,0 +1,841 @@
+            let Self {
+                client,
+                ctx,
+                continuation,
+            } = self;
+            Iter {
+                f,
+                client,
+                ctx,
+                continuation,
+            }
+        };
+
+        let stream = stream::try_unfold(
+            init,
+            move |Iter {
+                      f,
+                      client,
+                      ctx,
+                      continuation,
+                  }| {
+                Box::pin(async move {
+                    let continuation = match continuation {
+                        Continuation::None => None,
+                        Continuation::Continue(continuation) => Some(continuation),
+                        Continuation::Stop => return Ok::<_, RequestFailure>(None),
+                    };
+                    let (items, UnfoldNextState { ctx, continuation }) = f(UnfoldArgs {
+                        client: client.clone(),
+                        ctx,
+                        continuation,
+                    })
+                    .await?;
+                    Ok(Some((
+                        items,
+                        Iter {
+                            f,
+                            client,
+                            ctx,
+                            continuation: Continuation::from(continuation),
+                        },
+                    )))
+                })
+            },
+        );
+
+        stream
+            .map_ok(|items| stream::iter(items.into_iter().map(Ok)))
+            .try_flatten()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::client::Client;
+    use crate::client::ConnectionSettings;
+    use crate::client::InitializationError;
+    use crate::client::Request as _;
+    use crate::compile_model;
+    use crate::defs;
+    use crate::defs::*;
+    use crate::fga;
+    use crate::model::Check;
+    use crate::model::Relation;
+
+    fn setup_tracing() {
+        tracing_subscriber::fmt()
+            .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+            .without_time()
+            .pretty()
+            .try_init()
+            .expect("tracing should setup successfully");
+    }
+
+    macro_rules! test_client {
+        () => {
+            Client::try_new_store(
+                stdext::function_name!()
+                    .split("::")
+                    .filter(|x| *x != "{{closure}}")
+                    .collect::<Vec<_>>()
+                    .join("-"),
+                ConnectionSettings {
+                    address: "localhost".to_owned(),
+                    port: 8080,
+                    reset: true,
+                },
+            )
+            .await
+            .expect("Failed to initialize client")
+        };
+    }
+
+    #[tokio::test]
+    async fn test_try_init() {
+        setup_tracing();
+        let client = Client::try_with_store(
+            "lol".to_owned(),
```

Comment:
this store only exists on my machine, to change...

Reply:
removed the test: if `try_with_store` has a regression, other tests will fail

141120c68